test/fuzz/transformation_swap_function_variables_test.cpp

```diff
@@ -0,0 +1,125 @@
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %24 = OpVariable %7 Function
+         %25 = OpVariable %9 Function
+         %26 = OpVariable %11 Function
+         %27 = OpVariable %14 Function
+         %28 = OpVariable %7 Function
+         %29 = OpVariable %7 Function
+         %30 = OpVariable %7 Function
+         %32 = OpVariable %9 Function
+         %34 = OpVariable %11 Function
+         %36 = OpVariable %14 Function
+         %38 = OpVariable %7 Function
+         %40 = OpVariable %7 Function
+         %31 = OpLoad %6 %24
+               OpStore %30 %31
+         %33 = OpLoad %8 %25
+               OpStore %32 %33
+         %35 = OpLoad %10 %26
+               OpStore %34 %35
+         %37 = OpLoad %13 %27
+               OpStore %36 %37
+         %39 = OpLoad %6 %28
+               OpStore %38 %39
+         %41 = OpLoad %6 %29
+               OpStore %40 %41
+         %42 = OpFunctionCall %2 %22 %30 %32 %34 %36 %38 %40
+               OpReturn
+               OpFunctionEnd
+         %22 = OpFunction %2 None %15
+         %16 = OpFunctionParameter %7
+         %17 = OpFunctionParameter %9
+         %18 = OpFunctionParameter %11
+         %19 = OpFunctionParameter %14
+         %20 = OpFunctionParameter %7
+         %21 = OpFunctionParameter %7
+         %23 = OpLabel
+               OpReturn
+               OpFunctionEnd
+)";
+
+  // I've decleared a SPIR-V 1.5 latest version accroding to libspir.h
+  const auto env = SPV_ENV_UNIVERSAL_1_5;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // #ifndef NDEBUG
+  //   // Can't swap variable with itself.
+  //   ASSERT_DEATH(TransformationSwapFunctionVariables(7, 7).IsApplicable(
+  //                    context.get(), transformation_context),
+  //                "two ids are equal");
+  // #endif
+
+  // Can't swap instuction id for one of them not exits.
+  ASSERT_FALSE(TransformationSwapFunctionVariables(1, 2).IsApplicable(
+      context.get(), transformation_context));
+  // Can't swap instuctions one of them not OpVariable.
+  ASSERT_FALSE(TransformationSwapFunctionVariables(5, 24).IsApplicable(
+      context.get(), transformation_context));
+  //     Can't swap two instructions from two different blocks.
+  ASSERT_FALSE(TransformationSwapFunctionVariables(16, 26).IsApplicable(
+      context.get(), transformation_context));
+
+  // Successful transformations
+  {
+    //   Swap two OpVariable instructions in the same function
+    ASSERT_TRUE(TransformationSwapFunctionVariables(24, 28).IsApplicable(
+        context.get(), transformation_context));
+  }
+}
```

Comment:
You have called this test `NotApplicable`. In that case, you should not include the case where the transformation *is* applicable.

So, either change the name of the test, or have a separate test for the applicable case.

The case where the transformation is applicable should also apply it, and then assert that the transformation does the expected thing - i.e., that the transformed code has the expected syntax. See the tests that @Vasniktel has pointed you to above for ideas.

Reply:
done